internal/gatewayapi/tls.go

```diff
@@ -14,26 +14,29 @@ import (
 )
 
 // validateTLSSecretData ensures the cert and key provided in a secret
-// is not malformed and can be properly parsed.
-func validateTLSSecretData(secret *corev1.Secret) error {
-
+// is not malformed and can be properly parsed. It also returns the public key
+// encryption algorithm used to encrypt secret information.
+func validateTLSSecretData(secret *corev1.Secret) (string, error) {
+	var publicKeyAlgorithm string
```

Comment:
this API feels a little overloaded, but guessing `publicKeyAlgorithm` won't be needed in the future
can you raise a GH issue to support multiple certs for SNI based selection